File: src/main/java/com/developersam/game/ten/ClientMove.java
package com.developersam.game.ten;

/**
 * A data class that represents a client move.
 */
@SuppressWarnings("WeakerAccess")
public final class ClientMove {
    /**
     * Specifies the board before the human move.
     */
    final BoardData boardBeforeHumanMove = new BoardData();
    /**
     * Completes the picture by providing a human's move.
     */
    final Move humanMove = Move.DUMMY_MOVE;
    
    /**
     * Disable construction.
     */
    private ClientMove() {}
    
}

File: src/main/kotlin/com/developersam/game/ten/ClientMove.kt
package com.developersam.game.ten

/**
 * A data class that represents a client move.
 *
 * @property boardBeforeHumanMove specifies the board before the human move.
 * @property humanMove completes the picture by providing human's move in a tuple.
 */
class ClientMove private constructor(
        val boardBeforeHumanMove: BoardData, private val humanMove: IntArray
) {

    /**
     * [asMove] returns the move.
     */
    val asMove: Move get() = Move(a = humanMove[0], b = humanMove[1])

}
